Pioneer DDJ-400

nschloe opened the pull request:
After #2143 and #2312, this is another take on the Pioneer DDJ-400.

This PR implements most of the basic functionality of the controller:

  * Mixer Section (Faders, EQ, Filter, Gain, Cue)
  * Browsing and loading
  * Jogwheels, Scratching, Bending
  * Beat Sync
  * Beatjump mode
  * Hot cue Mode
  * Beat Loop Mode

I've successfully used this on two occasions and only found minor glitches. For example, when a hot cue point is set it lights up (this is good), but when shift is pressed, the lights go out. The functionality is still there though.

There are still a bunch of advanced controls missing, for example the entire effects section. This can be added in second PR though.

The visual XML composer was of great help creating this PR as was @WarkerAnhaltRanger first pass.

I'd suggest we keep it simple here and try to add more functionality in follow-up PRs. I'd be happy about some feedback of course.

uklotzde committed:
Add TrackLoader

uklotzde committed:
Add comments about immediate/deferred export of metadata

mi01 committed:
Traktor Kontrol S2MK3: Initial mapping script

mi01 committed:
Traktor Kontrol S2MK3: VuMeter handling

mi01 committed:
Traktor Kontrol S2MK3: Microphone button

mi01 committed:
Traktor Kontrol S2MK3: Samplers

mi01 committed:
Traktor Kontrol S2MK3: Remove HID specs from source file

mi01 committed:
Traktor Kontrol S2MK3: Added shift-functions for play, cue and sync

mi01 committed:
Traktor Kontrol S2MK3: Avoid spamming bus with VuMeter

mi01 committed:
Traktor Kontrol S2MK3: Better sampler handling

mi01 committed:
Traktor Kontrol S2MK3: Improved track selection & bugfixes

mi01 committed:
Traktor Kontrol S2MK3: Reverse and slip mode

mi01 committed:
Traktor Kontrol S2MK3: Better beatjump and loop handling

mi01 committed:
Traktor Kontrol S2MK3: GRID and preperation button

mi01 committed:
Traktor Kontrol S2MK3: JS code refactoring

uklotzde committed:
Move export of track metadata from TrackCollection into Library

uklotzde committed:
Get or add single tracks via TrackCollection

uklotzde committed:
Hide tracks via TrackCollection

uklotzde committed:
Reduce visibility of member functions in TrackDAO

uklotzde committed:
Fix naming conflict(?) and minor issues in DlgTagFetcher

AppVeyor build on Windows with VS 2017 failed.

mi01 committed:
Traktor Kontrol S2MK3: Fix review problems and remove raw data sending

uklotzde committed:
Delete obsolete member

uklotzde committed:
Delete undefined member function

uklotzde committed:
Lift enum into global namespace

ronso0 committed:
LateNight: fix separators in mini decks

ronso0 committed:
LateNight: paint Play buttons yellow when stopped deck plays from Cue / hotcue, use 'cue_default' for mini decks' Play right-click

ronso0 committed:
LateNight: use horizontal battery icon

ronso0 committed:
Merge branch 'master' of https://github.com/mixxxdj/mixxx into latenight-sliders-knobs

ronso0 committed:
LateNight: small fixes

ronso0 committed:
LateNight: show Fx main controls in one row when there's enough h-space in the window

ronso0 committed:
LateNight: smaller Reverse & Cue button, straight 2x4 hotcue grid

daschuer committed:
Guard also against qint64 overflow

ronso0 committed:
LateNight: simpler transport row layout

ronso0 committed:
LateNight: allwo to hide master/headphone mixer

ronso0 committed:
LateNight: clean up EQ knob templates

ronso0 committed:
LateNight test: increase hotcue labels in overview, contrast too

ronso0 committed:
LateNight: prepare color scheme creation

ronso0 committed:
LateNight: fix deck transport row spacing & alignment

ronso0 committed:
LateNight: fix launch image

ronso0 committed:
LateNight: prepare color scheme creation pt3

uklotzde committed:
Hide public functions in TrackDAO

uklotzde committed:
Reduce visibility of members in base classes

ronso0 committed:
LateNight: prepare color scheme creation pt4

ronso0 committed:
lATEnIGHT :: new fLIP theme

uklotzde committed:
Merge branch 'master' of git@github.com:mixxxdj/mixxx.git into dev_library

ronso0 committed:
LateNight: simplify spinny layout

ronso0 committed:
LateNight: adapt Flip theme

uklotzde committed:
Fix order of construction

ronso0 committed:
LateNight: simplify layout & qss styles

ronso0 committed:
LateNight: adapt Flip scheme pt2

ronso0 committed:
LateNight: clean up, sharpen spinny indicator

ronso0 committed:
LateNight: adapt Flip scheme pt4

ronso0 committed:
LateNight: whoopsy, restore lauch image

ronso0 committed:
LateNight: remove duplicate spinny container ObjectName

ronso0 committed:
LateNight: add btn_, knobs_ & sliders_SCHEME, further clean-up

ronso0 committed:
LateNight: adapt Flip scheme pt5

uklotzde committed:
Update src/library/trackcollectionmanager.h

Co-Authored-By: Be <be.0@gmx.com>

uklotzde committed:
Improve documentation of asychronous invocation function

uklotzde committed:
Don't pass QPointer around

uklotzde committed:
Merge branch 'master' of git@github.com:mixxxdj/mixxx.git into dev_library

uklotzde committed:
Fix typo

uklotzde committed:
Fix typo (again)

uklotzde committed:
Rename function and slot in TrackLoader

uklotzde committed:
Merge branch 'master' of git@github.com:mixxxdj/mixxx.git into dev_library

ronso0 committed:
LateNight: final layout clean-up, hopefully...

ronso0 committed:
LateNight: adapt Flip scheme pt6

uklotzde committed:
Add redundant type information

ronso0 committed:
LateNight: happy birthday Flip scheme!

uklotzde committed:
Move TrackLoader to target thread upon construction

uklotzde committed:
Use QMetaObject::invokeMethod() with functor

uklotzde committed:
Fix code formatting

uklotzde committed:
Use QStringLiteral for string constants

uklotzde committed:
Use parented_ptr

uklotzde committed:
Pass constructor arg as const-ref

uklotzde committed:
Don't std::move UserSettingsPointer

uklotzde committed:
Don't std::move QSharedPointer

uklotzde committed:
Add lost comment in GlobalTrackCache

uklotzde committed:
Revert parented_ptr for TrackCollectionManager in main window

Manual deletion is required to enforce the correct order of destruction.
This commit also fixes a double free bug caused by introducing
parented_ptr.

uklotzde committed:
Remove anonymous namespace from parented_ptr header

uklotzde committed:
Prevent unintended invocation of delete on parented_ptr

ronso0 committed:
LateNight: qss clean up, polish

ronso0 committed:
LateNight: balance mini spinny, add comments

ronso0 committed:
LateNight: keep spinny indicator shadow within spinny mask

ronso0 committed:
LateNight: update version number, restrict QLabel & QPushButton styles, clean up comments

ronso0 committed:
Merge branch 'master' of https://github.com/mixxxdj/mixxx into latenight-sliders-knobs

ronso0 committed:
LateNight: use default font size for skin settings buttons

ronso0 committed:
LateNight: decrease 8-samplers separator

ronso0 committed:
LateNight: bold font for spinboxes as well

ronso0 committed:
LateNight: remove skin settings category separator

ronso0 committed:
LateNight: restore BPM label in preview deck

uklotzde committed:
Fix QMetaObject::invokeMethod() for Qt < 5.10

uklotzde committed:
Delete unneeded parentheses

uklotzde committed:
Move captured variable into lambda scope

uklotzde committed:
Forward lambda parameters

uklotzde committed:
Add missing include directives

nschloe committed:
add Pioneer DDJ 400 midi mapping

nschloe committed:
add Pioneer DDJ 400 midi mapping

nschloe committed:
pioneer ddj400: add hotcue, beatloop support for deck 2

nschloe committed:
pioneer ddj400: add hotcue, beatloop support for deck 2

nschloe committed:
pioneer ddj400: add beatjump to deck 2

nschloe committed:
pioneer ddj400: add beatjump to deck 2

nschloe committed:
pioneer ddj400: leds for beat loops

nschloe committed:
pioneer ddj400: leds for beat loops

nschloe committed:
pioneer ddj400: add headphone cue buttons

nschloe committed:
pioneer ddj400: add headphone cue buttons

nschloe committed:
pioneer ddj400: eq for deck 2

nschloe committed:
pioneer ddj400: eq for deck 2

nschloe committed:
pioneer ddj400: get level meters to work

nschloe committed:
pioneer ddj400: get level meters to work

uklotzde committed:
Fix forwarding of captured parameters in lambda expressions

uklotzde committed:
Merge branch 'master' of git@github.com:mixxxdj/mixxx.git into dev_library

ronso0 committed:
LateNight: restore label colors, syncronise library button colors

ronso0 committed:
WPushButton POWERWINDOW: longpress temp toggle also for OFF state

uklotzde committed:
Update class comment

Holzhaus committed:
scripts: Add qsscheck.py script

Holzhaus committed:
.travis.yml: Run qsscheck.py on Linux CI builds

ronso0 committed:
LateNight: fix AutoDJ enable button bg color

Holzhaus committed:
.travis.yml: Cleanup travis config and run linter as separate job

Holzhaus committed:
.travis.yml: Add job names

Holzhaus committed:
.travis.yml: Remove unused before_script for osx builds

Holzhaus committed:
.travis.yml: Print QTDIR on osx builds

Holzhaus committed:
.travis.yml: Use python 3.7 by default on OSX

Holzhaus committed:
.travis.yml: Replace "matrix" alias with more descriptive "jobs"

Holzhaus committed:
.travis.yml: Fix setting global env variables

Holzhaus committed:
Revert ".travis.yml: Use python 3.7 by default on OSX"

This reverts commit 65724d57c0c143ab5e961afdf99c0348d7a8aea1.

Holzhaus committed:
.travis.yml: Only install python3 on qsscheck

ronso0 committed:
use emitValue for consistency

Holzhaus committed:
scripts/qsscheck: Pre-compile regexes

Holzhaus committed:
scripts/qsscheck: Reduce indention depth using continue statement

Holzhaus committed:
scripts/qsscheck: Only run glob-matching for actual glob objectnames

Holzhaus committed:
scripts/qsscheck: Improve qscheck script and add -p parameter

Holzhaus committed:
scripts/qsscheck: Move core logic from main() into separate function

Holzhaus committed:
scripts/qsscheck: Fix line exceeding maximum line lengths

Holzhaus committed:
scripts/qsscheck: Add docstrings for all functions

mi01 committed:
Traktor Kontrol S2MK3: Forums link in XML

mi01 committed:
Traktor Kontrol S2MK3: Do not stop already expired SYNC timer

mi01 committed:
Traktor Kontrol S2MK3: More precise intervals per rev

Co-Authored-By: Jan Holthuis <holthuis.jan@googlemail.com>

ronso0 committed:
LateNight: consolidate font styles, fix some alignment issues, finalize theme colors

ronso0 committed:
Merge branch 'master' into latenight-sliders-knobs

ronso0 committed:
LateNight: add Star icon for QuickEffect button

Holzhaus committed:
skin/legacyskinparser: Move some inline style hacks to default.qss

ronso0 committed:
LateNight: fix title/artist layout, clean up comments & css

Holzhaus committed:
widget/wtracktableview: Use lowercase for Auto DJ Queue locations

This makes wtracktableview use the same case as baseplaylistfeature,
baseexternallibraryfeature and controlpickermenu.

Holzhaus committed:
library: Remove duplicated PlaylistDAO::sendToAutoDJ

Holzhaus committed:
library/baseexternallibraryfeature: Add action to replace Auto DJ queue

Holzhaus committed:
library/baseplaylistfeature: Add action to replace Auto DJ queue

Holzhaus committed:
[aA]ddToAutoDJ instead of [sS]endToAutoDJ consistently in code base

ronso0 committed:
LateNight: fix labels in preview deck

ronso0 committed:
LateNight: tweak launch logo

uklotzde committed:
Explicitly connect/disconnect external track collections

uklotzde committed:
Merge branch 'master' of git@github.com:mixxxdj/mixxx.git into dev_library

uklotzde committed:
Hide internal drag&drop utility functions

uklotzde committed:
Prepare integration of external track collections

nschloe committed:
add jog wheels support

nschloe committed:
add jog wheels support

nschloe committed:
ddj-400: some documentation

nschloe committed:
ddj-400: some documentation

nschloe committed:
code factor issues

nschloe committed:
code factor issues

nschloe committed:
one more code factor

nschloe committed:
one more code factor

Be-ing commented:
> when a hot cue point is set it lights up (this is good), but when shift is pressed, the lights go out.

You may need to send a secondary MIDI message for the shifted state to prevent this. It is probably the same MIDI values as the controller sends when that button is pressed with shift.

Be-ing commented:
Thanks for continuing to work on this. Can you update the wiki page with whatever changes you have made?

> There are still a bunch of advanced controls missing, for example the entire effects section. This can be added in second PR though.

The mapping should be complete before merging. Having a mapping included in Mixxx and listing it as supported on the wiki should be an indication to (potential) users that it just works. Otherwise we set users up for disappointment.

nschloe commented:
> The mapping should be complete before merging.

For real? Then we can immediately cancel this. I won't be able to to get a complete mapping of the DDJ-400 up, this is just _so_ much work for me. I think, however, if we communicate clearly that not everything is working, it should be fine. There are multiple options for this:
  * not listing the DDJ-400 as supported
  * listing it at partially or basically supported
  * ...

ronso0 committed:
LateNight: add scheme previews

Be-ing commented:
What is still left to do? Is it only the effects?

ronso0 committed:
LateNight: remove playing_from_cue indicator

ronso0 committed:
LateNight: remove playing_from_cue indicator from sampler & mini decks, too

Holzhaus commented:
> What is still left to do? Is it only the effects?

https://github.com/mixxxdj/mixxx/pull/2385/files#diff-6a3e4882503838847f2b99ee50f272e0R19-R33

nschloe commented:
> What is still left to do? Is it only the effects?

No, it's more than that. For example, the DDJ-400 has a fancy-schmancy shift button that enables all kind of functionality: "keyboard mode", key shift, etc. I consider this _advanced_ and I probably won't ever need any of that, so I didn't bother with it right now. Also, the first PR #2143 got turned down because it took so long for it to get reviewed that the submitted had lost interest. I wanted to avoid this here.

But, like I said, I'd be happy to develop this further in a separate PR.

daschuer committed:
Merge pull request #2365 from uklotzde/dev_library

Extract TrackCollectionManager from Library

ronso0 committed:
LateNight: use displayValue instead of value for Play button background

Be-ing commented:
I think we are talking about different ideas of "complete". I don't care about reimplementing everything from #2143. I'd also be okay with leaving the alternate pad modes (keyboard, pad FX (whatever that does), key shift) unmapped. I do care that everything on the controller does something useful that makes sense with Mixxx, which could be fairly different from how it works with Rekordbox. So how about mapping the samplers and effects, fixing the known bugs, and calling that good enough?

Be-ing commented:
Here is a proposal for how to map the unusual effects section on this controller:
  * Beat `<` / `>` buttons: focus previous/next effect
  * FX select button: unfocus; make no effect focused
  * 1/2/Master switch: assign effect unit 1 to deck 1/2/master as labeled
  * Level/Depth knob: metaknob of focused effect; if no effect is focused, controls the mix knob for effect unit 1
  * On/Off button: enable/disable focused effect

nschloe commented:
Okay, I'll keep working on it here then.

> You may need to send a secondary MIDI message for the shifted state to prevent this.

I don't understand what you mean here.

Be-ing commented:
I see that the hotcue buttons send MIDI messages with a 1st byte (status) as 0x97 when pressed but it is 0x98 when shift is pressed. You only send outgoing MIDI for 0x97, but not for 0x98. You may need to send outgoing MIDI for 0x98 to prevent the LEDs from flickering when shift is pressed.

Be-ing committed:
Merge pull request #2342 from ronso0/latenight-sliders-knobs

LateNight :: Redesign part 2

Holzhaus committed:
Merge branch 'master' of https://github.com/mixxxdj/mixxx into qsscheck

nschloe commented:
Another thing I don't understand about development: mixxx always appears to create a file `DDJ-400_MIDI_1.midi.xml` in `~/.mixxx/controllers/`. 
```
ls -l ~/.mixxx/controllers/
```
```
-rw-r--r-- 1 nschloe nschloe  51K Dec  8 22:19 DDJ-400_MIDI_1.midi.xml
-rw-r--r-- 1 nschloe nschloe 527K Nov 26 17:20 lodash.mixxx.js
-rw-r--r-- 1 nschloe nschloe  48K Nov 26 16:40 midi-components-0.0.js
lrwxrwxrwx 1 nschloe nschloe   77 Nov 26 16:32 Pioneer-DDJ-400.midi.xml -> /home/nschloe/software/mixxx/nschloe/res/controllers/Pioneer-DDJ-400.midi.xml
lrwxrwxrwx 1 nschloe nschloe   78 Nov 26 16:32 Pioneer-DDJ-400-script.js -> /home/nschloe/software/mixxx/nschloe/res/controllers/Pioneer-DDJ-400-script.js
```
It seems to be a copy of the actual `Pioneer-DDJ-400.midi.xml`. This has caught be off-guard a couple of times, editing `Pioneer-DDJ-400.midi.xml` and wondering why nothing changes. Is this a bug?

Be-ing commented:
It's not exactly a bug; it's more of a confusing workflow that could be improved. The "DDJ-400_MIDI_1.midi.xml" is automatically copied by Mixxx and the filename comes from the MIDI device name. That is the file Mixxx automatically loads on startup by seeing that there is a MIDI device which has the same name. Maybe we can rethink this together with the coming changes for controllers in 2.4.

nschloe commented:
> I see that the hotcue buttons send MIDI messages with a 1st byte (status) as 0x97 when pressed but it is 0x98 when shift is pressed. You only send outgoing MIDI for 0x97, but not for 0x98. You may need to send outgoing MIDI for 0x98 to prevent the LEDs from flickering when shift is pressed.

The "flickering" happens _before_ 0x98 is sent. You press a hotcue, 0x97 is sent, mixxx sends a message to turn on the light, all fine. Then when shift is pressed, without ever touching a hotcue button, the lights go out.

nschloe commented:
Another question: Is there any way of filtering the the controllerDebug midi codes? Now that blinking Cue is working, it's hard to find the messages I'm actually looking for in the sea of cue-on/cue-off messages.

Be-ing commented:
Right, what it seems like is happening is that the controller firmware maintains separate states for the LEDs depending on whether shift is pressed. So if you want to avoid flickering when shift is pressed, both the controller's shifted and unshifted states need to be set whenever hotcue_X_enabled changes.

nschloe commented:
> It's not exactly a bug; it's more of a confusing workflow that could be improved. The "DDJ-400_MIDI_1.midi.xml" is automatically copied by Mixxx and the filename comes from the MIDI device name. That is the file Mixxx automatically loads on startup by seeing that there is a MIDI device which has the same name. Maybe we can rethink this together with the coming changes for controllers in 2.4.

I'll file a bug for it so we don't lose track.

nschloe commented:
> Right, what it seems like is happening is that the controller firmware maintains separate states for the LEDs depending on whether shift is pressed.

Indeed.

> So if you want to avoid flickering when shift is pressed, both the controller's shifted and unshifted states need to be set whenever hotcue_X_enabled changes.

I don't understand "setting the state". I press the shift button and the controller is in shift mode; what to change about that?

Holzhaus committed:
skins/LateNight: Remove unused WText declaration from style.qss

Be-ing commented:
Whenever hotcue_X_enabled changes, Mixxx needs to send MIDI messages for both the shifted and unshifted MIDI channels (status bytes 0x97 and 0x98). So try copying and pasting all the outputs for the buttons and changing 0x97 to 0x98.

The Hercules P32 controller that I previously used also has this quirky behavior. That's why I added the `sendShifted` feature in the Components library.

nschloe commented:
> So try copying and pasting all the outputs for the buttons and changing 0x97 to 0x98.

That works! One needs to have
```
            <output> 
                <group>[Channel1]</group>
                <key>hotcue_1_enabled</key>
                <status>0x97</status>
                <midino>0x00</midino>
                <minimum>0.5</minimum>
            </output>
            <output>
                <group>[Channel1]</group>
                <key>hotcue_1_enabled</key>      
                <status>0x98</status>
                <midino>0x00</midino>
                <minimum>0.5</minimum>
            </output>
```
side by side.

Be-ing commented:
Please add a comment to the XML file explaining why that is necessary.

Holzhaus committed:
Merge pull request #2387 from Holzhaus/latenight-qss-fix

skins/LateNight: Remove unused WText declaration from style.qss

Holzhaus committed:
Merge branch 'master' of https://github.com/mixxxdj/mixxx into qsscheck

mi01 committed:
Traktor Kontrol S2MK3: Use toggleControl()

Be-ing commented:
I have a better idea for the effects. Instead of scrolling forward/back for focusing effects:

  * Level/Depth knob: metaknob of focused effect; if no effect is focused, controls the mix knob for effect unit 1
  * Beat `<`: focus effect 1
  * Beat `>`: focus effect 2
  * FX select button: focus effect 3
  * On/Off button: enable/disable focused effect
  * 1/2/Master switch: assign effect unit 1 to deck 1/2/master as labeled

Unfocus by pressing the button for the focused effect again (so if effect 1 is focused, press Beat `<` to unfocus and make the knob control the mix knob of the unit).

nschloe committed:
ddj-400: transfer over more from waker

nschloe committed:
ddj-400: transfer over more from waker

nschloe committed:
make sure the hotcue lights stay ON when pressing shift

nschloe committed:
make sure the hotcue lights stay ON when pressing shift

nschloe committed:
remove some faulty outputs

nschloe committed:
remove some faulty outputs

ronso0 committed:
LateNight: fix Play buttons

ronso0 committed:
LateNight: removed unused icons from Flip scheme

nschloe committed:
beat jump

nschloe committed:
beat jump

nschloe committed:
ddj400: version bump to 2.3

nschloe committed:
ddj400: version bump to 2.3

nschloe committed:
js style

nschloe committed:
js style

nschloe committed:
parseInt(x, 10) -> parseInt(x)

nschloe committed:
parseInt(x, 10) -> parseInt(x)

nschloe committed:
more code style

nschloe committed:
more code style

nschloe committed:
ddj400: handle hotcue by intrinsics

nschloe committed:
ddj400: handle hotcue by intrinsics

nschloe committed:
ddj400: reloop_toggle -> reloop_andstop

nschloe committed:
ddj400: reloop_toggle -> reloop_andstop

nschloe commented:
Okay, I've merged more stuff from the original PR now, and more things are working now. I still need some assistance with the lights going on on shift-press though. The hotcues are fixes, but other buttons are also losing light, e.g., beat sync. How to prevent this?

Be-ing commented:
I presume the same way. Send output for the shifted MIDI channel in addition to the unshifted MIDI channel.

Be-ing commented on res/controllers/Pioneer-DDJ-400.midi.xml:
```diff
@@ -506,7 +506,7 @@
                 <description>RELOOP/EXIT (DECK1) - press - (loop off) Reloop, (loop on) Loop exit</description>
                 <group>[Channel1]</group>
-                <key>reloop_toggle</key> <!-- check if correct -->
+                <key>reloop_andstop</key> <!-- check if correct -->
```
Why did you change this? I think `reloop_andstop` makes sense for the shifted press, but I don't understand why you are using it for the unshifted press too.

nschloe replied:
By mistake. Fixed now.

nschloe committed:
code factor

nschloe committed:
code factor

nschloe committed:
fix reloop_toggle

nschloe committed:
fix reloop_toggle

Be-ing committed:
Merge pull request #2388 from ronso0/latenight-fixes2

LateNight fixes

nschloe committed:
code factor

nschloe committed:
code factor

luzpaz committed:
Fix various typos

Found via `codespell -q 3 -I ../mixxx-whitelist.txt -S *.ts,*.po,*.rtf,./.git,./src/library,./lib,./build/wix`

Be-ing committed:
Merge pull request #2390 from luzpaz/typos

Fix various typos

uklotzde committed:
Merge pull request #2373 from ronso0/pushbutton_longpress

allow longpress temporary toggle also for OFF state of POWERWINDOW mode

uklotzde committed:
Merge branch 'master' of git@github.com:mixxxdj/mixxx.git into library-autodj-fixes

# Conflicts:
#	src/library/baseplaylistfeature.cpp
#	src/library/baseplaylistfeature.h
#	src/widget/wtracktableview.cpp

uklotzde committed:
Merge pull request #2235 from daschuer/lp1839669

fix assertion in Lp1839669

mi01 committed:
Traktor Kontrol S2MK3: Sampler pregain knob

Holzhaus committed:
Merge pull request #2375 from Holzhaus/qsscheck

Add qsscheck to CI pipeline and improve .travis.yml config

Holzhaus committed:
scripts/qsscheck: Also check default stylesheets

uklotzde committed:
Merge branch 'master' of git@github.com:mixxxdj/mixxx.git into dnd_cleanup

uklotzde committed:
Fix function signatures

uklotzde committed:
Rename function/signal/slot parameter

uklotzde committed:
Fix indentation/whitespaces

uklotzde committed:
Send a single signal after modifying multiple playlists

uklotzde committed:
Merge pull request #2383 from Holzhaus/library-autodj-fixes

Library AutoDJ cleanup

uklotzde committed:
Fix Travis CI build for macOS

uklotzde committed:
Fix Travis CI build for macOS

luzpaz committed:
Fix typos (mixx->mixxx)

Fixes various cosmetic source comment typos. Similar relevant mixco typos have been merged upstream: https://github.com/arximboldi/mixco/commit/3dce9410e7ccd7a0144e579023cad6d1c700473e

uklotzde committed:
Merge pull request #2395 from luzpaz/mixx-to-mixxx

Fix typos (mixx->mixxx)

Be-ing committed:
Merge pull request #2386 from uklotzde/dnd_cleanup

Hide internal drag&drop utility functions

uklotzde committed:
Prevent temporary copies of QList

Be-ing committed:
Merge pull request #2396 from uklotzde/travis_macos

Fix Travis CI build for macOS

uklotzde committed:
Merge branch '2.2' of git@github.com:mixxxdj/mixxx.git

# Conflicts:
#	.travis.yml

uklotzde committed:
Iterate over const QList to avoid cloning

uklotzde committed:
Delete unused local variable

uklotzde committed:
Wrap long lines

uklotzde committed:
Rename pointer variable

uklotzde committed:
Merge pull request #2382 from Holzhaus/legacy-style-hacks-to-qss

skin/legacyskinparser: Move some inline style hacks to default.qss

Holzhaus committed:
Merge pull request #2391 from Holzhaus/qsscheck

scripts/qsscheck: Also check default stylesheets

mi01 committed:
Traktor Kontrol S2MK3: Implement review feedback

mi01 committed:
Traktor Kontrol S2MK3: Comment on LED color

mi01 committed:
Traktor Kontrol S2MK3: Better SYNC button handling

mi01 committed:
Traktor Kontrol S2MK3: Better loop handling

mi01 committed:
Traktor Kontrol S2MK3: Set LEDs on startup according to current control state

mi01 committed:
Traktor Kontrol S2MK3: Better slip mode handling

mi01 committed:
Traktor Kontrol S2MK3: Fix beatloop roll

Be-ing committed:
Merge pull request #2348 from mi01/TraktorKontrolS2MK3-Mapping

Native Instruments Traktor Kontrol S2 MK3 mapping

Be-ing committed:
Merge remote-tracking branch 'upstream/2.2'

Holzhaus commented:
Are you using any features new in Mixxx 2.3 in this mapping? If not, can you retarget this PR to the 2.2 branch and rebase on 2.2? This will make it possible to include this in 2.2.4 instead of 2.3.

Holzhaus commented:
Are you using any features new in Mixxx 2.3 in this mapping? If not, can you retarget this PR to the 2.2 branch and rebase on 2.2? This will make it possible to include this in 2.2.4 instead of 2.3.

uklotzde committed:
Fix function call

Be-ing committed:
Merge pull request #2401 from uklotzde/unhide_track

Don't hide tracks when adding to crates

daschuer committed:
Merge pull request #2393 from uklotzde/purge_tracks_from_playlists

lp1845837: Speed up purging of tracks

uklotzde committed:
Merge branch '2.2' of git@github.com:mixxxdj/mixxx.git

# Conflicts:
#	src/library/baseplaylistfeature.cpp
#	src/library/dao/playlistdao.cpp
#	src/library/playlistfeature.h
#	src/library/playlisttablemodel.cpp
#	src/library/setlogfeature.h

uklotzde committed:
Merge branch 'master' of git@github.com:mixxxdj/mixxx.git into dev_load_tracks

uklotzde committed:
Add missing #include directive for qAsConst

uklotzde committed:
Add missing #include directive for qAsConst

uklotzde committed:
Merge branch 'master' of git@github.com:mixxxdj/mixxx.git into dev_load_tracks

uklotzde committed:
Add missing #include directive for qAsConst

uklotzde committed:
Merge branch 'master' into dev_load_tracks

daschuer committed:
Merge pull request #2397 from uklotzde/dev_load_tracks

Refactor loading of tracks from database

nschloe committed:
Merge branch 'pioneer-ddj400' of github.com:nschloe/mixxx into pioneer-ddj400

Be-ing commented:
This PR contains a lot of unrelated commits now. It looks like you rebased onto master instead of 2.2. Do an interactive rebase `git rebase -i mixxxdj/2.2` to rebase on 2.2 and remove all commits that are not related to this mapping.

nschloe commented:
>  It looks like you rebased onto master instead of 2.2.

I did indeed. I don't know how to undo either. `git rebase -i mixxxdj/2.2` just gives `fatal: invalid upstream 'mixxxdj/2.2'`. The easiest thing is probably simply to start a new PR.

Be-ing commented:
Have you set up the mixxxdj repository as a remote? The [wiki](https://mixxx.org/wiki/doku.php/using_git#set_up_git) has instructions to set it up as a remote called "upstream", so the command would be `git rebase -i upstream/2.2`

nschloe commented:
Yeah I always to that, but then I've never done an "interactive" rebase. I don't know what do to with the interface git gives me there.

Be-ing commented:
Delete the lines for every commit that doesn't have to do with your mapping, leaving only the commits for your mapping.

nschloe commented:
> Delete the lines for every commit that doesn't have to do with your mapping, leaving only the commits for your mapping.

There are hundreds of commits, and I only see the title, not the author. It's too much work. I'll simply start a new PR.

nschloe commented:
See #2403.

Be-ing commented:
All of your commits are at the end. It should be easy to delete the commits before that.

nschloe commented:
I've played around with it for half an hour now but I can't seem to get the rebase to work. I did
```
$ git remote add upstream https://github.com/mixxxdj/mixxx.git
$ git fetch upstream
$ git rebase -i upstream/2.2
$ git push -f origin pioneer-ddj400
```
but I still have all those commits here. Gah.
